src/irmin-pack/pack.mli

```diff
@@ -60,14 +63,17 @@ end
 module type MAKER = sig
   type key
 
+  type index
```

Comment:
Exposing an `index` type here felt clumsy, but I'm unsure about how otherwise to propagate the constraint that `File` gives a `MAKER` with the correct index type.